### PR TITLE
add mergeAll to (Z)STM companion object

### PR DESCRIPTION
**Pull request by sken77**
https://github.com/zio/zio/issues/2802

### CODE REVIEW

File: core/shared/src/main/scala/zio/stm/ZSTM.scala [hunk 1]
```diff
@@ -1085,6 +1085,14 @@ object ZSTM {
    */
   val none: STM[Nothing, Option[Nothing]] = succeedNow(None)
 
+  /**
+   * Merges an `Iterable[IO]` to a single IO, working sequentially.
```
Comment (mijicd) [inline core/shared/src/main/scala/zio/stm/ZSTM.scala]: `IO` should be replaced with `ZSTM`.

File: core/shared/src/main/scala/zio/stm/STM.scala [hunk 2]
```diff
@@ -178,9 +178,9 @@ object STM {
   /**
    * @see See [[zio.stm.ZSTM.mergeAll]]
    */
-  def mergeAll[R, E, A, B](
-    in: Iterable[ZSTM[R, E, A]]
-  )(zero: B)(f: (B, A) => B): ZSTM[R, E, B] = ZSTM.mergeAll(in)(zero)(f)
+  def mergeAll[E, A, B](
+    in: Iterable[ZSTM[Any, E, A]]
```
Comment (mijicd) [inline core/shared/src/main/scala/zio/stm/STM.scala]: `STM[E, A]`

File: core/shared/src/main/scala/zio/stm/STM.scala [hunk 3]
```diff
@@ -178,9 +178,9 @@ object STM {
...
+  )(zero: B)(f: (B, A) => B): ZSTM[Any, E, B] = ZSTM.mergeAll(in)(zero)(f)
```
Comment (mijicd) [inline core/shared/src/main/scala/zio/stm/STM.scala]: Same here.

Comment (sken77) [reply core/shared/src/main/scala/zio/stm/STM.scala]: that was sloppy sorry about that

Comment (mijicd) [reply core/shared/src/main/scala/zio/stm/STM.scala]: No worries :)

**Comment by sken77** [comment]
I changed the implementation of `mergeAll` to use `zipWith` instead of zip

**Pull request merged**